### ISSUE TITLE
sd3.01  Create Git Plugins SSH Credentials

### ISSUE THREAD

**Issue by rgleason** [Rasbats/shipdriver_pi#385]
This preparation, done once. 
https://opencpn-manuals.github.io/main/AlternativeWorkflow/InstallConfigure/GithubPreps.html
and these paragraphs should be moved and integrated into [GIT_REPO ](https://opencpn-manuals.github.io/main/AlternativeWorkflow/InstallConfigure/BuilderEnv.html#_git_repo) in my opinion, because these steps must be done first.

At the beginning add:

` 2. Create a fork of https://github.com/OpenCPN/plugins.git on Github.`

Move items  [3 to  8](https://opencpn-manuals.github.io/main/AlternativeWorkflow/InstallConfigure/GithubPreps.html) to  [GIT_REPO ](https://opencpn-manuals.github.io/main/AlternativeWorkflow/InstallConfigure/BuilderEnv.html#_git_repo)

3. You will need the ssh url to your OpenCPN/plugins fork, something like git@github.com:Rasbats/plugins.git. This can be found in the Github UI:`

thru

`8. Take a note about GIT_REPO and GIT_KEY_PASSWORD, they are used in later configuration steps.`

**Comment by leamas** [comment]
I don't agree. 

GIT_REPO is part of the Builder Environment reference page, here are no steps whatsoever, and there should not be.

Furthermore, the reason to have this in the very beginning is that the values of GIT_REPO and GIT_KEY_PASSWORD are established here.  These are used in next step where the  builders are configured.  Part of this configuration is the environment.

I'm more than open to ideas to move everything about git-push to the end. But when we have tried that, it means coming back to builder configuration after the git-push stuff. So far,  I think this was worse. Could certainly be wrong.

**Comment by Rasbats** [comment]
The only change that might help is to move _Builder environment_ to the top of the list, above _Appveyor_. Otherwise I agree with Alec.

**Comment by rgleason** [comment]
Yes, I agree that the GIT_REPRO and GIT_KEY_PASSWORD have to be First established and in order to do that, you must Create a "plugins" repository and then create the key and password.  It is sequential and done once for all plugins.  

I see that this is at the top under [CI Services Configuration](https://opencpn-manuals.github.io/main/AlternativeWorkflow/InstallConfigure.html)  and yes this is in the right sequential order.

I note that Githubpreps.html could be revised somewhat because there are **_one time instructions_** and there are **_routinely_** used github commands.   I think you should move the **_one-time instructions_** to the top and title the **_Routine_** commands accordingly.

I think it would be clearer if the "[Builder Environment"](https://opencpn-manuals.github.io/main/AlternativeWorkflow/InstallConfigure/BuilderEnv.html)  which is needed for all the Builders was directly under "Cloudsmith"  or at the top of the "Builders" list.

**Comment by leamas** [comment]
We could certainly be clearer about the git-push configuration being one-time, agreed. Will look into it.

It is also possible to move the Builder Environment page to the beginning of the builder configuration. But to keep the flow,  it must then be rewritten to step-by-step form rather than the current reference form.  That is. we will be adding something like "define configuration values" as a  first step to the builder configuration. And re-phrase it accordingly.

Is this what we want?

**Comment by Rasbats** [comment]
The _Builder Environment_ page needs a small edit. GIT_KEY_PASSWORD section needs a back reference to _Github OpenCPN/plugins fork_.

Put this on my TODO list! None of this is critical IMHO.

**Comment by rgleason** [comment]
This all sounds very good to me.  Do you want this closed?

**Comment by Rasbats** [comment]
Please leave as a reminder. Thanks.
